### PR TITLE
Picard ConcatenateMetricsFiles

### PR DESCRIPTION
**Pull request by jasonwalker80**
initial revision of ConcatenateMetricsFiles.pm

### CODE REVIEW

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

File: lib/perl/Genome/Model/Tools/Picard/ConcatenateMetricsFiles.pm [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            is_many => 1,
+        },
+        output_file  => {
+            is  => 'String',
+            doc => 'The output concatenated metrics file',
+        },
+        subclass_name => {
+            is => 'String',
+            is_optional => 1,
+            valid_values => ['CollectRnaSeqMetrics', 'GenotypeConcordance', 'MarkDuplicates'],
```
Comment (tmooney) [inline lib/perl/Genome/Model/Tools/Picard/ConcatenateMetricsFiles.pm]: There are a few other Picard subclasses that have `_metric_header_as_key` defined:
* `CollectAlignmentSummaryMetrics`
* `CollectGcBiasMetrics`
* `CollectInsertSizeMetrics`

I'm not sure if this tool would be useful for any of them?

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

File: lib/perl/Genome/Model/Tools/Picard/ConcatenateMetricsFiles.pm [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        $parser_class_name .= '::'. $self->subclass_name;
+    }
+
+    my @metrics_files = sort $self->metrics_files;
+
+    my %metrics;
+    for my $metrics_file (@metrics_files) {
+         my $metrics_hash_ref = $parser_class_name->parse_file_into_metrics_hashref($metrics_file);
+         if (defined($metrics{$metrics_file})) {
+             $self->fatal_error_message('Duplicate metrics file: '. $metrics_file);
```
Comment (tmooney) [inline lib/perl/Genome/Model/Tools/Picard/ConcatenateMetricsFiles.pm]: This should just be `fatal_message`.

File: lib/perl/Genome/Model/Tools/Picard/ConcatenateMetricsFiles.pm [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    my @headers = keys %{$first_metrics_hash_ref};
+
+    my $writer = Genome::Utility::IO::SeparatedValueWriter->create(
+        output => $self->output_file,
+        separator => $self->separator,
+        headers => \@headers,
+        ignore_extra_columns => $self->ignore_extra_columns,
+    );
+    unless ($writer) {
+        $self->fatal_error_message('Unable to open output writer for file: '. $self->output_file);
```
Comment (tmooney) [inline lib/perl/Genome/Model/Tools/Picard/ConcatenateMetricsFiles.pm]: This should just be `fatal_message`.

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

File: lib/perl/Genome/Model/Tools/Picard/ConcatenateMetricsFiles.pm [hunk 4]
```diff
@@ -0,0 +1,83 @@
+package Genome::Model::Tools::Picard::ConcatenateMetricsFiles;
+
+use strict;
+use warnings;
+
+use Genome;
+use Data::Dumper;
```
Comment (tmooney) [inline lib/perl/Genome/Model/Tools/Picard/ConcatenateMetricsFiles.pm]: Is this using `Data::Dumper`?

**Pull request merged**